google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/PubSubExample.java

```diff
@@ -1,665 +0,0 @@
-   *
-   * @see <a href="https://cloud.google.com/pubsub/docs/access_control">Access Control</a>
-   */
-  private static class TestSubscriptionPermissionsAction extends TestPermissionsAction {
-    @Override
-    public void run(PubSub pubsub, Tuple<String, List<String>> param) throws Exception {
-      String subscription = param.x();
-      List<String> permissions = param.y();
-      List<Boolean> booleanPermissions =
-          pubsub.testSubscriptionPermissions(subscription, permissions);
-      System.out.printf("Caller permissions on subscription %s%n", subscription);
-      for (int i = 0; i < permissions.size(); i++) {
-        System.out.printf("%s: %b%n", permissions.get(i), booleanPermissions.get(i));
-      }
-    }
-  }
-
-  static {
-    CREATE_ACTIONS.put("topic", new CreateTopicAction());
-    CREATE_ACTIONS.put("subscription", new CreateSubscriptionAction());
-    INFO_ACTIONS.put("topic", new TopicInfoAction());
-    INFO_ACTIONS.put("subscription", new SubscriptionInfoAction());
-    LIST_ACTIONS.put("topics", new ListTopicsAction());
-    LIST_ACTIONS.put("subscriptions", new ListSubscriptionsAction());
-    DELETE_ACTIONS.put("topic", new DeleteTopicAction());
-    DELETE_ACTIONS.put("subscription", new DeleteSubscriptionAction());
-    GET_IAM_ACTIONS.put("topic", new GetTopicPolicyAction());
-    GET_IAM_ACTIONS.put("subscription", new GetSubscriptionPolicyAction());
-    REPLACE_IAM_ACTIONS.put("topic", new AddIdentityTopicAction());
-    REPLACE_IAM_ACTIONS.put("subscription", new AddIdentitySubscriptionAction());
-    TEST_IAM_ACTIONS.put("topic", new TestTopicPermissionsAction());
-    TEST_IAM_ACTIONS.put("subscription", new TestSubscriptionPermissionsAction());
-    ACTIONS.put("create", new ParentAction(CREATE_ACTIONS));
-    ACTIONS.put("info", new ParentAction(INFO_ACTIONS));
-    ACTIONS.put("list", new ParentAction(LIST_ACTIONS));
-    ACTIONS.put("delete", new ParentAction(DELETE_ACTIONS));
-    ACTIONS.put("pull", new ParentAction(PULL_ACTIONS));
-    ACTIONS.put("get-policy", new ParentAction(GET_IAM_ACTIONS));
-    ACTIONS.put("add-identity", new ParentAction(REPLACE_IAM_ACTIONS));
-    ACTIONS.put("test-permissions", new ParentAction(TEST_IAM_ACTIONS));
-    ACTIONS.put("replace-push-config", new ReplacePushConfigAction());
-  }
-
-  private static void printUsage() {
-    StringBuilder actionAndParams = new StringBuilder();
-    for (Map.Entry<String, PubSubAction> entry : ACTIONS.entrySet()) {
-      actionAndParams.append("\n\t").append(entry.getKey());
-
-      String param = entry.getValue().params();
-      if (param != null && !param.isEmpty()) {
-        actionAndParams.append(' ').append(param.replace("\n", "\n\t\t"));
-      }
-    }
-    System.out.printf("Usage: %s [<project_id>] operation [entity] <args>*%s%n",
-        PubSubExample.class.getSimpleName(), actionAndParams);
-  }
-
-  @SuppressWarnings("unchecked")
-  public static void main(String... args) throws Exception {
```

Comment:
We might want to consider keeping this file; it shows how to write a command-line app which calls PubSub. The command line options could all be preserved; the calls to the pubsub client would need to be converted. This could also be a good place to point people to who need to migrate existing code. 

You can address this in a separate PR.